modules/elasticsearch_connector_views/src/Plugin/views/query/ElasticsearchViewsQuery.php

```diff
@@ -124,6 +133,9 @@ public static function create(ContainerInterface $container, array $configuratio
                         ->get('elasticsearch_connector_views');
     $plugin->setLogger($logger);
 
+    $entity_type_manager = $container->get('entity_type.manager');
+    $plugin->setEntityTypeManager($entity_type_manager);
```

Comment:
This is sort of atypical, why add and use a setter, why not do the usual thing with a create method, which is to implement a constructor, and pass the services/arguments to set as members of the class to it?

Reply:
I know, but this plugin is a bit strange and I did not want to change the architecture while implementing tests so I followed what was there already to add the dependency.

Reply:
We could come back to this in a new PR and use the constructor to inject dependencies.